### ISSUE TITLE
Post 2.6 - Start cutting all new versions as releases

### ISSUE THREAD

**Issue by DanVanAtta** [triplea-game/triplea#10585]
@DanVanAtta said in PR#10425

> I am strongly considering as well that following the 2.6 launch, any merge to master would result in a new release, and not a new prerelease. I mention this to speak to how I think every change will need to be done in an incremental fashion (I'd hope that no branch lives longer than a week or two), and that releases would be far more frequent. I do not have a mindset where I think releases ought to be large, monolithic and incompatible jumps, I want quite the opposite.

@TheDog-GH said: 

> Please dont do the above, there will be hundreds of versions on the Players computers, the forum mods will be saying upgrade all the time.

> As a player I would prefer to have a **stable yearly regular update**, say in January, and not have to wait years for the next round of goodies/improvements.

> Consider a new player, downloads a buggy version and is totally put off, they would be lost to us.

Apologies for digressing, but I feel strongly about this.

_Originally posted by @TheDog-GH in https://github.com/triplea-game/triplea/issues/10425#issuecomment-1146766754_

**Comment by DanVanAtta** [comment]
@TheDog-GH 
>  there will be hundreds of versions on the Players computers,

I agree, though why is this a problem if they are all compatible with each other?

>  the forum mods will be saying upgrade all the time.

Why would they be? Why couldn't the software do this? Are you presuming that most of these versions would have problems?

> As a player I would prefer to have a stable yearly regular update, say in January, and not have to wait years for the next round of goodies/improvements.

The goal/intent was to have two 'stable' releases each year. That has not happened.
Second, the concept of 'stable' & 'unstable' releases I believe is an anti-pattern altogether and is a shortcoming for "we can't help but break our code". The solution is not to aggregate all broken changes into a large burst of "okay, we now fix all this stuff after the fact", but instead to invest heavily in not breaking the code in the first place. The code should just simply be stable. If we cannot keep the code 'stable', our efforts to make it 'stable' are really questionable.

> Consider a new player, downloads a buggy version and is totally put off, they would be lost to us.

Knowing that any update is going to be a release will hopefully be more incentive to be very diligent that merged code is stable.

**Comment by DanVanAtta** [comment]
@TheDog-GH bump ^

**Comment by TheDog-GH** [comment]
@DanVanAtta
I was waiting for other peoples comments, as I dont have too much to add.

> Knowing that any update is going to be a release will hopefully be more incentive to be very diligent that merged code is stable.

This could also put Devs off, because they have to more diligent, recently Im thinking of

- 2.6 unzipping of maps, that sometimes did not work as intended. (Roger went back to 2.5, luckily I had copies of the zips)
- map.yml format that has changed, causing maps not to load between 2.6 testing

Most players are shielded from the above unless they are a tester.

.
**Perhaps you should canvas opinions on the forums by doing a poll, the current verses your proposed with its benefits?**
People dont like change, so its the margin of votes I will be looking at.

**Comment by DanVanAtta** [comment]
Forum votes are always odd as:
1. The opinionated views are the ones that tend to respond
2. The votes can trickle in over a very long time, which is not conducive for moving forward
3. It's hard to know how well informed the votes are, whether they are knee jerk reactions or actually well thought out positions

> This could also put Devs off, because they have to more diligent,

I'm okay with this. Any 'less diligent' efforts and it is up to the maintainers to fix those gaps. I think a counter-argument could be made as Devs appreciate their changes seeing the light of day and being released sooner rather than later.

> Players will have to keep downloading the current version, as it can be out of date within a day, they might get bored of constantly doing this.

When we change the version number, that is when the "recommended version" would change. It's always been intended to have a release version go out every 3 to 6 months (which is not happening, so if we don't go with the proposal to cut all versions as releases, then we need something else to solve that problem). So no, players would in no way be forced to stay with the very latest version.

> Perhaps you should canvas opinions on the forums by doing a poll, the current verses your proposed with its benefits?

Perhaps, I kinda wish we could have one place to discuss this project and not have to split the conversation so often. To another extent, this is a technical decision; it's only important to identify concerns so we can think about how we can avoid potential problems. Otherwise we need to know a way where we can solve the problem of infrequent releases and how we can create a better forcing function to always maintain releasable software.

**Comment by TheDog-GH** [comment]
Im sort of coming round, but still have reservations as to have how it will be received by the forum.

But moving forward lets try it, we can always go back to how it is now and that's not ideal.
